lib/bucket_files.js

```diff
@@ -30,15 +30,32 @@ module.exports = (client, bucket, logger) => {
 
   const current = async () => {
     try {
-      logger.info(`executing listObjects for bucket: ${bucket}`)
-      const results = await client.listObjects({ Bucket: bucket }).promise()
-      logger.debug(`listObjects response for bucket: ${bucket}`, results)
-      logger.info(`listObjects returned ${results.Contents.length} for bucket: ${bucket}`)
-      return results.Contents
+        var params = { Bucket: bucket };
+        var allObjectsRead = false;
+        var objectsRead = new Array();
+        logger.info(`executing listObjects for bucket: ${bucket}`);
+        while (!allObjectsRead) {
+            var results = await client.listObjectsV2(params).promise();
+            results.Contents.forEach(element => {
+                // Could delete non-essential properties from the data here
+                // For example: element.StorageClass = null ;
+                //
+                // Could lower the memory footprint by converting the Timestamp to UTC millis
+                //              element.LastModified = new Date(element.LastModified).getTime();
+                objectsRead.push(element);
+            });
+            if (results.IsTruncated)
+                params.ContinuationToken = results.NextContinuationToken;
+            else
+                allObjectsRead = true;
+        }
+        logger.debug(`listObjects response for bucket: ${bucket}`, results);
+        logger.info(`listObjects returned ${objectsRead.length} for bucket: ${bucket}`)
+        return objectsRead
     } catch (err) {
-      logger.debug(`listObjects returned error`, err)
-      const message = errors.format(err)
-      throw new Error(message)
+        logger.debug(`listObjects returned error`, err)
```

Comment:
Can you re-format the code to use two rather than four spaces? This matches the existing style.